pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpression.java

```diff
@@ -117,6 +118,9 @@ class FunctionCall implements RexExpression {
     // whether the function is a distinct function.
     @ProtoProperties
     private boolean _isDistinct;
+    // the list of RexExpressions that represents the pre-args to the function.
+    @ProtoProperties
+    private List<RexExpression> _functionPreArgs;
```

Comment:
Trying to understand it better, why can't we reuse `_functionOperands` here for literals?

Reply:
correct, it will break the query plan, e.g. intermediate stage result reference.
So need to apply `_functionPreArgs` to `_functionOperands` during the query execution phase.

Reply:
I don't see the usage of this field. Am I missing something?